src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -81,6 +83,9 @@ def generate_argv(
         args.append(f"--config-file={mypy.config}")
     if python_version:
         args.append(f"--python-version={python_version}")
+    # @TODO: appeared in 0.700, is that OK?
+    args.append("--skip-cache-mtime-check")
```

Comment:
Need to bail if version is lower

Reply:
Parsing the version using `from packaging.version import Version` should be fairly easy, so probably worth doing.

AFAICT, you only need to disable the flag, not the cache itself (it's harmless to create, you'll just miss it).

Reply:
This is how we do it for MyPy Protobuf 

https://github.com/pantsbuild/pants/blob/14e01228487f5a757941cf04756b09af9d6dd799/src/python/pants/backend/codegen/protobuf/python/rules.py#L110-L116

I think it's this way to accommodate using forks of MyPy?

Reply:
I also turned OFF writing the cache on lower versions, because its a waste (thanks @Eric-Arellano for the idea)